### PR TITLE
core/macros: add math helper macros

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This adds a few commonly useful math helper macros to `core/macros`:

 - `SIGNOF()` returns the sign of an integer
 - `DIV_ROUND()` performs arithmetic integer division
 - `DIV_ROUND_UP()` performs arithmetic integer division, always rounding to the next integer larger in magnitude. 

### Testing procedure

`unittests/tests-core` was extended with test cases for the new macros

### CODE REVIEW

**Comment by kaspar030** [comment]
... and they are supposed to be used in compile-time situations, I guess?

**Comment by benpicco** [comment]
Also at run-time.
It's just that I've found myself open-coding those way too often, so I thought providing a macro would be appropriate.

**Comment by maribu** [comment]
https://github.com/RIOT-OS/RIOT/blob/6db9960741cc93acbc903f482a93391333f8b6e8/cpu/stm32/periph/ptp.c#L63-L71

> It's just that I've found myself open-coding those way too often, so I thought providing a macro would be appropriate.

Yes, indeed. Is just one of the cases where this would have been useful.

File: core/include/macros/math.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * @brief Returns the sign of @p a, either -1 or 1
+ */
+#define SIGNOF(a) _Generic(a, unsigned char: 1,          \
+                              unsigned short: 1,         \
+                              unsigned int: 1,           \
+                              unsigned long: 1,          \
+                              unsigned long long: 1,     \
+                              default: __signof(a))
```
Comment (maribu) [inline core/include/macros/math.h:45]:
```suggestion
                              default: (a) < 0 ? -1 : 1)
```

? `__singof()` is only used here anyway and this way `clangd` won't nag me about reserved identifiers being used.

**Review by maribu — APPROVED** [review]

**Pull request merged**